gms/gossiper.cc

```diff
@@ -2234,7 +2237,7 @@ future<> gossiper::stop() {
 }
 
 bool gossiper::is_enabled() const {
-    return _enabled;
+    return _enabled && !_abort_source.abort_requested();
```

Comment:
I don't see why this patch is necessary. Gossiper never ends up with an inconsistent state, even if listeners failed. The same state which we have on listener failure can already be observed by RPCs, before the listener fails.

Reply:
We can consider it separately, as it is indeed out of scope.
I just don't see a reason to process any gossip state changes after abort is requested.